orttraining/orttraining/training_api/include/lr_scheduler.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "orttraining/training_api/include/optimizer.h"
+
+namespace onnxruntime {
+namespace training {
+namespace api {
+
+/**
+ * @brief Base class for learning rate scheduler.
+ */
+struct LRSchedulerBase {
+ public:
+  LRSchedulerBase(Optimizer& optimizer) : optim_(optimizer) {}
+  virtual ~LRSchedulerBase() = default;
+
+  /**
+   * @brief Compute learning rate taking step and initial learning rate as inputs, then update
+   * the adapted learning rate into optimizer.
+   *
+   * This should be called every time optimizer update states related to learning rate calculations,
+   * for example, initial learning and step.
+   */
+  Status Step() {
+    return optim_.SetLearningRate(ComputeLearningRateInternal());
+  }
+
+ protected:
+  int64_t GetStepInternal() {
+    return optim_.optimizer_state_.step;
+  }
+
+  float GetInitialLRInternal() {
+    return optim_.optimizer_state_.initial_lr;
+  }
+
+ private:
+  virtual float ComputeLearningRateInternal() = 0;
+  Optimizer& optim_;
```

Comment:
Fixed the first comment. For the second one, I think we would suggest user to create pointers if they want to do that way, while we need improve anyway on how we expose LRSchduler (and even Base) later to users, because ideally we should allow user to define their own lr schduler.

Reply:
My first and second comment were kind of implying the same thing. If we require a shared pointer in LRScehduler, then the compiler can create a copy constructor for us for scheduler and so doing what I showed above would work.
Thanks for making the change.